### PR TITLE
#7 Update change logs

### PR DESCRIPTION
**Pull request by Heaghneyn**

### CODE REVIEW

**Comment by Heaghneyn** [comment]
- [ ] add logging of new viruses and new circulating

File: R/utils.R [hunk 1]
```diff
@@ -624,18 +630,46 @@ run_single_table_diagnostic <-
 #' @param log_file str: location of cache file
 #' @param .time dttm: time of update
 #' @param .user double: user who conducted the action
+#' @param .event_type str: START, PROCESS, INFO, ERROR, ALERT, END
 #' @param .event str: event to be logged
 #' @returns Return true if cache updated
 update_polis_log <- function(log_file = Sys.getenv("POLIS_LOG_FILE"),
                              .time = Sys.time(),
                              .user = Sys.getenv("USERNAME"),
+                             .event_type = "START",
```
Comment (nish-kishore) [inline R/utils.R:639]: You could leave the default value empty which would force people to put in an event type ever time. That way we don't have accidental "START" entries

Comment (Heaghneyn) [reply R/utils.R:639]: but don't we not want people to put in values since this is all supposed to run behind the scenes?

File: R/utils.R [hunk 2]
```diff
@@ -172,7 +174,8 @@ get_table_data <- function(api_key = Sys.getenv("POLIS_API_Key"),
           ": ",
           table_size,
           " new or updated records identified!"
-        )
+        ),
+        .event_type = "START"
```
Comment (nish-kishore) [inline R/utils.R:178]: I think this should be a "PROCESS" since we are using "START" and "END" exclusively to identify full runs

File: R/utils.R [hunk 3]
```diff
@@ -1227,6 +1289,101 @@ get_env_site_data <- function(){
...
+#'
+#' @import dplyr readr
+#' @param log_file str: location of POLIS log file
+#' @param polis_data_folder str: location of the POLIS data folder
+
+archive_log <- function(log_file = Sys.getenv("POLIS_LOG_FILE"),
+                        polis_data_folder = Sys.getenv("POLIS_DATA_CACHE")){
+
+  #create log archive
+  if(dir.exists(file.path(polis_data_folder, "Log_Archive")) == FALSE){
```
Comment (nish-kishore) [inline R/utils.R:1358]: Not something you need to change but I'd prefer our standard to stop testing booleans. So `dir.exists` returns a boolean so you can just do `!dir.exists()` to get the opposite, I can make the edits for now

File: R/utils.R [hunk 4]
```diff
@@ -3698,7 +3937,7 @@ preprocess_cdc <- function(polis_data_folder = Sys.getenv("POLIS_DATA_CACHE")) {
   cli::cli_process_done()
 
   cli::cli_process_start("Processing and cleaning AFP/non-AFP files")
-  afp.files.01 <- list.files(path = paste0(polis_data_folder, "/Core_Ready_Files/"), pattern = "^(afp_linelist_2001-01-01_2023).*(.rds)$", full.names = TRUE)
+  afp.files.01 <- list.files(path = paste0(polis_data_folder, "/Core_Ready_Files/"), pattern = "^(afp_linelist_2001-01-01_2024).*(.rds)$", full.names = TRUE)
```
Comment (nish-kishore) [inline R/utils.R:3940]: Let's change this to be more dynamic in naming 2024. Otherwise we'll have to dig and find this every time...

**Review by nish-kishore — CHANGES_REQUESTED** [review]
Awesome work, some very minor comments. Let me know if you want to chat about it

File: R/init.R [hunk 5]
```diff
@@ -192,11 +193,18 @@ get_polis_data <- function(type = "all"){
...
+
     sapply(tables, function(x) get_table_data(.table = x))
 
   }
 
   if(type == "pop"){
+
+    update_polis_log(.event = "Start POLIS pop download",
+                     .event_type = "START")
+
```
[the remark this answers is not in the document]
Comment (Heaghneyn) [reply R/init.R:207]: added end for pop download

[eb305ac](https://github.com/nish-kishore/tidypolis/pull/17/commits/eb305ac3cd64777a94778471c7604d2917f5e78e)

File: R/utils.R (hunk 3 above)
Comment (Heaghneyn) [reply R/utils.R]: fixed double call to boolean

[afa5603](https://github.com/nish-kishore/tidypolis/pull/17/commits/afa5603652edcb33ec47363da614e948f74e97c7)

**Review by nish-kishore — APPROVED** [review]

**Pull request merged**